src/platform/i386/kernel.c

```diff
@@ -68,14 +72,19 @@ kern_memory_setup(struct multiboot *mb, u32_t mboot_magic)
 	for (i = 0 ; i < mb->mods_count ; i++) {
 		struct multiboot_mod_list *mod = &mods[i];
 
-		printk("\t- %d: [%08x, %08x)", i, mod->mod_start, mod->mod_end);
+		char *cmdline = (char *)mod->cmdline;
+		int cmdline_len = strnlen((const char*)cmdline, CMDLINE_MAX_LEN);
+		assert(cmdline_len >= CMDLINE_REQ_LEN);
+		int addr_offset = cmdline_len - CMDLINE_REQ_LEN;
+
+		printk("\t- %d: [%08x, %08x) : %s", i, mod->mod_start, mod->mod_end, mod->cmdline);
 		/* These values have to be higher-half addresses */
 		glb_memlayout.mod_start = chal_pa2va((paddr_t)mod->mod_start);
 		glb_memlayout.mod_end   = chal_pa2va((paddr_t)mod->mod_end);
 
-		glb_memlayout.bootc_vaddr = (void*)hextol((char*)mod->cmdline);
-		assert(((char*)mod->cmdline)[8] == '-');
-		glb_memlayout.bootc_entry = (void*)hextol(&(((char*)mod->cmdline)[9]));
+		glb_memlayout.bootc_vaddr = (void*)hextol((char *)(cmdline + addr_offset));
```

Comment:
Why did you replace the previous support for placement of the vaddr of the boot component based on the file name for the module with this?  Won't that break quite a bit?  won't that require a lot of info that the user doesn't have on the command line?